main.go

```diff
@@ -30,7 +30,8 @@ import (
 	"github.com/release-engineering/retrodep/retrodep"
 )
 
-const defaultTemplate string = "{{if .TopPkg}}{{.TopPkg}}:{{.TopVer}}/{{end}}{{.Pkg}}:{{.Ver}}"
+const defaultTemplate string = "{{if .TopPkg}}{{.TopPkg}}:{{if .TopVer}}{{.TopVer}}{{else}}vX.X.X{{end}}" +
+	"/{{end}}{{.Pkg}}:{{if .Ver}}{{.Ver}}{{else}}vX.X.X{{end}}"
```

Comment:
I think I'd like something shorter than `vX.X.X`. Is `?` no good there?

Reply:
Not sure, but could this be a multi-line string?

Reply:
> Not sure, but could this be a multi-line string?

You mean with backticks? Yes, I guess it could. We want to avoid newlines in the template output, so we'll need to use space trimming. I also just saw that we could use the "or" template function to make it simpler, e.g.
```
const defaultTemplate = `
  {{- if .TopPkg -}}
    {{.TopPkg}}:{{or .TopVer "?"}}/
  {{- end -}}
  {{.Pkg}}:{{or .Ver "?"}}`
```

Reply:
Neat shortening with the `or` command. I guess we can remove the if for the toplevel package presence, as it is expected to be always present. Or we can use `{{or .TopPkg "UnidentifiedTopPackage"}}`. That would ensure to correctly identify a vendored package. But for that we would have to distinguish what is vendored and what is top-level only pkg (line).

How do you do the space trimming for such multiline string?

As for `vX.X.X` to `?`, I have no stronger preference. I just found `vX.X.X` more visually appealing.

Reply:
No, [Reference](https://godoc.org/github.com/release-engineering/retrodep/retrodep#Reference)'s TopPkg field will be empty if Pkg is the top-level package. That's how we arrange for packages reporting on only the top-level package to look different to those reporting on vendored dependencies.

The "{{- " and " -}}" bits do the [space-trimming](https://godoc.org/text/template#hdr-Text_and_spaces).

I quite like "?" as the 'unknown' value because it cannot be ether a valid semantic version or a valid tag name.

(While we're talking about the format, I would have preferred ";" as the separate between the top-level package detalis and the vendored package details, as "/" will appear in both names...)